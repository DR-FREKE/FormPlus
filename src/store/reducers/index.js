export { default as uiReducer } from "./ui.reducer";
export { default as templateReducer } from "./template.reducer";
