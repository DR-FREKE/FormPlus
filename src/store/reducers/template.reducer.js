import * as type from "../actions/action.types";

const initialState = {
  failed: false,
  searching: false,
  sorting: false,
  date_sorting: false,
  filtering: false,
  loading: true,
  total: 0,
};
const templateReducer = (state = initialState, action) => {
  switch (action.type) {
    case type.LOADING:
      return {
        ...state,
        loading: true,
        failed: false,
      };
    case type.GET_TEMPLATE:
      return {
        ...state,
        loading: false,
        form_template: action.payload.templates,
        original: action.payload.templates,
        total: action.payload.total,
      };
    case type.FILTER_TEMPLATE:
      return {
        ...state,
        loading: false,
        filtering: true,
        failed: false,
        sorting: false,
        date_sorting: false,
        filter_data: action.payload.filter_data,
        form_template: action.payload.result,
        original: action.payload.template,
        total: action.payload.total,
      };
    case type.SORT_BY_ORDER:
      return {
        ...state,
        loading: false,
        date_sorting: false,
        sorting: true,
        form_template: action.payload.result,
        total: action.payload.total,
      };
    case type.SORT_BY_DATE:
      return {
        ...state,
        loading: false,
        sorting: false,
        date_sorting: true,
        form_template: action.payload.result,
        total: action.payload.total,
      };
    case type.SEARCH_TEMPLATE:
      return {
        ...state,
        loading: false,
        filtering: false,
        searching: true,
        failed: false,
        form_template: action.payload.result,
        original: action.payload.template,
        total: action.payload.total,
      };
    case type.GET_ERROR:
      return {
        ...state,
        loading: false,
        searching: false,
        failed: true,
        failed_message: action.payload,
      };
    default:
      return state;
  }
};

export default templateReducer;
