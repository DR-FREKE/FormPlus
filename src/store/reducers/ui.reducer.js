import * as type from "../actions/action.types";

const uiReducer = (state = {}, action) => {
  switch (action.type) {
    case type.APP_MODE:
      return {
        ...state,
        dark_mode: action.paylaod,
      };
    default:
      return state;
  }
};

export default uiReducer;
