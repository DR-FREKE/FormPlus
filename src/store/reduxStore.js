import { createStore, applyMiddleware, compose } from "redux";
import { persistReducer, persistStore } from "redux-persist";
import storage from "redux-persist/lib/storage";
import thunk from "redux-thunk";
import rootReducer from "./indexReducer";

let persistConfig = {
  key: "root",
  storage,
};

let composeEnhancers = compose;

composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(thunk))
);

export const persistor = persistStore(store);
