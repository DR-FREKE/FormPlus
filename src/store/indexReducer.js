import { combineReducers } from "redux";
import * as reducer from "./reducers";

const reducers = combineReducers({
  mode: reducer.uiReducer,
  template: reducer.templateReducer,
});

export default reducers;
