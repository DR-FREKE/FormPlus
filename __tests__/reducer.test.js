import { templateReducer, uiReducer } from "../src/store/reducers";
import * as actions from "../src/store/actions/action.creator";

const initialState = {
  dark_mode: false,
};

describe("test the reducer", () => {
  it("should test for app mode", () => {
    expect(
      uiReducer(initialState, actions.appMode({ theme: false }))
    ).toMatchSnapshot();
  });

  it("should test for default state", () => {
    expect(uiReducer({}, actions.appMode)).toMatchSnapshot();
  });

  it("should test for templateReducer", () => {
    expect(templateReducer({}, actions.getTemplates)).toMatchSnapshot();
  });

  it("should return initial state for templateReducer", () => {
    expect(templateReducer({ loading: true }, actions.getTemplates)).toEqual({
      loading: true,
    });
  });
});
