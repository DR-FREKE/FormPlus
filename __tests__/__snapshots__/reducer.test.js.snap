// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the reducer should test for app mode 1`] = `
Object {
  "dark_mode": undefined,
}
`;

exports[`test the reducer should test for default state 1`] = `Object {}`;

exports[`test the reducer should test for templateReducer 1`] = `Object {}`;
